src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -184,39 +186,143 @@ public async Task EmissionsForecastsBatch_MissingRequiredParams_ReturnsBadReques
         Assert.That(result?.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
     }
 
-    [Test]
-    public async Task EmissionsForecastsBatch_SupportedDataSources_ReturnsOk()
+    [TestCase("2021-09-01T08:30:00Z", "2021-09-01T09:00:00Z", "2021-09-01T12:00:00Z", "eastus", 1, TestName = "EmissionsForecastsBatch expects OK for eastus 1 element")]
+    [TestCase("2021-09-01T08:30:00Z", "2021-09-01T08:30:00Z", "2021-09-02T08:30:00Z", "westus", 3, TestName = "EmissionsForecastsBatch expects OK for westus 3 element")]
+    public async Task EmissionsForecastsBatch_SupportedDataSources_ReturnsOk(DateTimeOffset reqAt, DateTimeOffset start, DateTimeOffset end, string location, int nelems)
     {
         IgnoreTestForDataSource("data source does not implement '/emissions/forecasts/batch'", DataSourceType.JSON);
 
         _dataSourceMocker.SetupBatchForecastMock();
+        var inputData = Enumerable.Range(0, nelems).Select(x => new EmissionsForecastBatchDTO() 
+        {
+            RequestedAt = reqAt,
+            DataStartAt = start,
+            DataEndAt = end,
+            Location = location
+        });
+
+        using (var result = await PostJSONBodyToURI(inputData, batchForecastURI))
+        {
+            Assert.That(result, Is.Not.Null);
+            Assert.That(result!.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(result!.Content, Is.Not.Null);
+            using (var data = await result!.Content.ReadAsStreamAsync())
```

Comment:
As a rule, we do not test for content in our integration tests because the test data has been too brittle.  

I recommend removing this part of the test to be consistent with that approach for now.  There is a story to address how we generate the mock data content.

Reply:
@bderusha , this is something that won't hurt given the fact that returning OK is not enough. At least getting the data and checking whether the content has something, imo is more valuable than nothing.